pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -133,7 +134,11 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 	var ephemeralStats []stats.VolumeStats
 	var persistentStats []stats.VolumeStats
 	for name, v := range metricVolumes {
-		metric, err := v.GetMetrics()
+		metric, err := func() (*volume.Metrics, error) {
+			trace := utiltrace.New("Volume stat calculation")
+			defer trace.LogIfLong(1 * time.Second)
+			return v.GetMetrics()
```

Comment:
This looks pretty neat. Now that we are only logging this information when something takes too long, did you wanted to add more information to the trace - such as volume name(which is readily available)?

Reply:
Add the volume name and pod namespace/name in the trace description.